lib/workos/directory_group.rb

```diff
@@ -25,6 +27,10 @@ def to_json(*)
       }
     end
 
+    def to_hash
```

Comment:
I went back and forth on `to_h` vs `to_hash` and eventually settled on `to_hash`, imitating the one on the `Hash` itself - https://apidock.com/ruby/Hash/to_hash (and because it's marginally more descriptive). But I don't have a strong opinion on this one way or another - happy to change to `to_h`, if you think that's better/more consistent.

As far as leaving it out altogether, I'm a little uncomfortable with that - if somebody was using a `Hash` method on the result, then all of a sudden, we don't really have support for it. This way, they at least have an easy work around. Counter-argument: we can just suggest that they call `to_json` as a workaround (which is what I'm doing here anyway), but that starts to feel a bit hacky. Also, converting the User or Group info into a Hash kinda makes sense to me - we start off with basically a Hash of data from the IdP, doesn't seem like that much of a stretch to give them a direct way to convert it back to a Hash?

Reply:
hmm, well `Hash` implements both `to_h` and `to_hash`, and between the two, it sounds like `to_h` is the more generally applicable method. But I don't feel strongly about this either, so go with your gut.

I'm still not convinced that we need to support `to_hash` going forward. Isn't it part of the `DeprecatedHash` API?

In either case, we can punt on making a decision here until we are ready to remove `DeprecatedHash` completely.

Reply:
👍 that's compelling enough :), went ahead and switched to using `to_h`. As far as whether we need it or not, we can't quite punt until removing `DeprecatedHash`, or at least not without potentially breaking users again 😭 - my current warning message suggests using `to_h` to get a Hash, if you're trying to use a Hash method on the results... so it wouldn't be great if we removed support for the method we were telling people to use 😅.

> Isn't it part of the `DeprecatedHash` API

I currently have the method independent of `DeprecatedHash` - `DirectoryGroup` and `DirectoryUser` both implement it, so I've not really been treating as a part of `DeprecatedHash`, as much as a backwards compatibility helper for anyone who really wants to be able to get the results as a Hash.

Reply:
Ah, good point.

I think there's two options here:

1. We consider `to_h` as deprecated, and remove it when we remove `DeprecatedHash`, since none of the other resource objects (`WorkOS::Profile`, etc.) implement a `to_h`. So then we would also want to update that warning message now, and avoid tell them to use `to_h`.
2. We keep `to_h` here, but then we also implement it on the rest of the resource objects as well.

So, my goal here is that we don't end up with a different kind of inconsistency in our API with `DirectoryUser` and `DirectoryGroup` being the only classes that implement `to_h`, 

Does that make sense?

My initial preference was for option 1 since it seemed like less work, but I would be fine with option 2 as well.

Reply:
Yeah, that 💯 makes sense... as for the options ... option 1 is definitely less work, but 🤔 I still feel bad for taking the Hash away after users have potentially been using it this whole time. So, I guess I prefer option 2...

Though that also commits us to implementing a `to_h` method any time we add a new class in the future, which is also annoying and liable to end up with inconsistent behavior if we forget to. So, I'm waffling on my preferences a bit.

Reply:
> Though that also commits us to implementing a to_h method any time we add a new class in the future, which is also annoying and liable to end up with inconsistent behavior if we forget to.

Probably wouldn't be a bad idea to refactor those classes, and maybe have them all inherit from a base class that provides `to_h`, etc.

But yeah, we'd have to commit to doing that work.

Reply:
👍 to a shared base class to include the `to_h` implementation... though it still doesn't prevent future classes from showing up that don't inherit from the base class 😭. But we can probably add tests and/or linters to try and catch that.

Either way, I think it's worth shipping this PR as it is now, then coming back with a separate one to add `to_h` to everything else - does that seem reasonable to you?

Reply:
> Either way, I think it's worth shipping this PR as it is now, then coming back with a separate one to add to_h to everything else - does that seem reasonable to you?

That sounds good! I think we are good to go here.